internal/task_documentation/task_documentation.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2022 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+const DirectoryPermissions = 0755
+
+var yamlDir = flag.String("yaml", "", "Location of the generated YAML files")
+var startPath = flag.String("startPath", "", "Location of path to start from")
+var taskDir = flag.String("taskDir", "task", "Location of tekton tasks to scrape")
```

Comment:
Do we need both `taskDir` and `startPath`? Why not just take one path parameter?

Reply:
My thought here was that if a `taskDir` is provided, we only start to walk from that subdirectory of the `startPath`. If not, we would walk from the root, collecting all Tekton tasks as we walk the entirety of the directory structure under the `startPath`.